webgui-new/src/pages/project.tsx

```diff
@@ -0,0 +1,104 @@
+import { SyntheticEvent, useContext } from 'react';
+import { Header } from 'components/header/header';
+import { AccordionMenu } from 'components/accordion-menu/accordion-menu';
+import { Box, CircularProgress, styled, Tab, Tabs } from '@mui/material';
+import { TabName } from 'enums/tab-enum';
+import { Diagrams } from 'components/diagrams/diagrams';
+import { Metrics } from 'components/metrics/metrics';
+import { GitDiff } from 'components/git-diff/git-diff';
+import { CodeMirrorEditor } from 'components/codemirror-editor/codemirror-editor';
+import { AppContext } from 'global-context/app-context';
+import { Credits } from 'components/credits/credits';
+import { Welcome } from 'components/welcome/welcome';
+import { UserGuide } from 'components/user-guide/user-guide';
+
+interface TabPanelProps {
+  children?: React.ReactNode;
+  index: number;
+  value: number;
+}
+
+const OuterContainer = styled('div')({
```

Comment:
This will be a general remark for the whole codebase.
I see you use this `styled()` function [provided by the MUI library](https://mui.com/system/styled/). My concern is that this way the styling is defined in the TypeScript code. I think separating the CSS styling into separate files would be a better, more robust and easier to maintain approach.

As I see, there is  [interoperability](https://mui.com/material-ui/guides/interoperability/) provided by the MUI library with e.g. CSS modules. Would it be possible to refactor the codebase to use CSS modules instead of inline CSS stylesets?

Reply:
The main advantage of using Material UI's `styled` function is that you can utilize the styles and colors declared in the `theme.ts` file, and the colors automatically update if the theme is switched to dark/light. So you don't have to check seperately that the theme is light or dark, you can just use for example `theme.colors.primary`, because it is defined for both light and dark modes using MUI's `createTheme()` (see `theme.ts`).

As this is not possible to do with regular CSS, I don't think it is a good idea to use CSS modules instead of MUI's built-in functions, as MUI styled components provide a better abstraction for styling, in my opinion.

However, I can move all of these elements created with `styled` to a separate file for each component, e.g. `styled-components.ts`, give them a more consistent name, and import them from there for each component. This way they would be better separated, like CSS modules. At some places, I used the `sx` prop for styling, which is the same as the inline `style` prop in CSS, but you can use all of MUI's themes and functionalities. In some cases, this is mandatory to use, like when states determine some of the styling, so these cannot be separated.

Reply:
While mixing TypeScript and CSS in such a way is a bit odd for me, I accept that it is the preferred and advised solution by MUI. The suggested separation to `styled-components.ts` files should be done though, so that at least there is a logical separation.

Reply:
I separated them to different files. For the pages, I created files in the `theme` folder and named them accordion to the page. (e.g. `project-styles.tsx`)